src/validation.cpp

```diff
@@ -5226,7 +5228,10 @@ void ChainstateManager::Reset()
 {
     m_ibd_chainstate.reset();
     m_snapshot_chainstate.reset();
-    m_active_chainstate = nullptr;
+    {
+        LOCK(::cs_main);
```

Comment:
Again, I think it's fine to hold this lock for the entirety of this function (in fact, the other ChainstateManager data members should be guarded by the same mutex).

Reply:
Done! Not 100% sure about the other members, we could do those in the future.